lib/_http_client.js

```diff
@@ -504,19 +504,17 @@ ClientRequest.prototype._deferToConnect = function(method, arguments_, cb) {
   // in the future (when a socket gets assigned out of the pool and is
   // eventually writable).
   var self = this;
```

Comment:
While you're here, `self` doesn't seem to be necessary. Mind removing it? It can be replaced with, as far as I can tell, a `var socket = this.socket`. The bottom reference to `self` can just be used as `this`.